demo/repo/docker-build.sh

```diff
@@ -0,0 +1,18 @@
+set -ex
+cd `dirname $0`
+
+# https://docs.docker.com/engine/userguide/eng-image/multistage-build/ would not be a good match here since docker-build does not support volumes, needed for caching the local repository between builds
+
+BUILD_IMG=examplecorp/spring-petclinic-build:$BUILD_TAG
+BUILD_CTR=spring-petclinic-build-$BUILD_TAG
+
+# Prepare the image in which we will build the app.
+docker build -t $BUILD_IMG -f app/Dockerfile.build app
+
+# Run a temporary container to build the app and grab petclinic.war out of it.
+trap "docker rm -f $BUILD_CTR; rm -f app/petclinic.war" EXIT
+docker run --name $BUILD_CTR -v m2repo:/m2repo $BUILD_IMG /src/build.sh
+docker cp $BUILD_CTR:/src/target/petclinic.war app
+
+# Build the actual application image.
+docker build -t localhost/examplecorp/spring-petclinic:$BUILD_TAG app
```

Comment:
Things are even simpler now with multi-staged docker builds, and this all script can be handled by a single Dockerfile.

Reply:
As I wrote elsewhere in the PR, I did actually try to do this with a multistage build, but soon found that I could not mount the Maven repository cache as a volume, which kills performance. Would be possible to prepopulate a cache as a base layer, though this does not easily allow for sharing caches across images. If you have any better ideas, let me know.

Reply:
My experience is that it's way simpler to setup a maven proxy, which will offer high-speed downloads from localhost

Reply:
Yes, that is an option, though you would need to do some work to make the proxy URL be defined as an `ENV`. For purposes of a demo like this, you would need to set up such a proxy as part of the demo I guess.

Reply:
> I could not mount the Maven repository cache as a volume

This has since been implemented in Buildkit.

Reply:
Also, Dagger would now be a reasonable choice: https://github.com/dagger/dagger/discussions/1677#discussioncomment-2424449